docs/modules/ROOT/pages/extensibility.adoc

```diff
@@ -0,0 +1,134 @@
+* `public`: part of the public API of a library
+* `internal`: subset of `public` that is either discouraged or potentially unsafe (e.g.
+`_transfer` on ERC20)
+* `external`: subset of `public` that is ready to be exported as-is by contracts (e.g.
+`transfer` on ERC20)
+* `storage`: storage variable functions
+
+Then:
+
+* Must implement `public` and `external` functions under a namespace
+* Must implement `private` functions outside the namespace to avoid exposing them
+* Must prefix `internal` functions with an underscore (e.g.
+`ERC20._mint`)
+* Must not prefix `external` functions with an underscore (e.g.
+`ERC20.transfer`)
+* Must prefix `storage` functions with the name of the namespace to prevent clashing with other libraries (e.g.
+`ERC20balances`)
+* Must not implement any `@external`, `@view`, or `@constructor` functions
+* Can implement initializers (never as `@constructor` or `@external`)
+* Must not call initializers on any function
+
+=== Contracts
+
+* Can import from libraries
+* Should implement `@external` functions if needed
+* Should implement a `@constructor` function that calls initializers
+* Must not call initializers in any function beside the constructor
+
+Note that since initializers will never be marked as `@external` and they won't be called from anywhere but the constructor, there's no risk of re-initialization after deployment.
+It's up to the library developers not to make initializers interdependent to avoid weird dependency paths that may lead to double construction of libraries.
+
+== Presets
+
+Presets are pre-written contracts that extend from our library of contracts.
+They can be deployed as-is or used as templates for customization.
+
+Some presets are:
+
+* https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/account/Account.cairo[Account]
+* https://github.com/OpenZeppelin/cairo-contracts/blob/main/tests/mocks/ERC165.cairo[ERC165]
+* https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/token/erc20/presets/ERC20Mintable.cairo[ERC20Mintable]
+* https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/token/erc20/presets/ERC20Pausable.cairo[ERC20Pausable]
+* https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/token/erc20/presets/ERC20Upgradeable.cairo[ERC20Upgradeable]
+* https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/token/erc20/presets/ERC20.cairo[ERC20]
+* https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/token/erc721/presets/ERC721MintableBurnable.cairo[ERC721MintableBurnable]
+* https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/token/erc721/presets/ERC721MintablePausable.cairo[ERC721MintablePausable]
+* https://github.com/OpenZeppelin/cairo-contracts/blob/main/src/openzeppelin/token/erc721/enumerable/presets/ERC721EnumerableMintableBurnable.cairo[ERC721EnumerableMintableBurnable]
```

Comment:
I would avoid links to `main`, they can/will break in the future. Not sure if there is a good alternative that won't become out of date too quickly though...